### PR TITLE
Improve performance of unescapeXML

### PR DESCRIPTION
**Pull request by mogsie**
Half of #120 fixed by this change.

### CODE REVIEW

File: test/escape-test.js [hunk 1]
```diff
@@ -49,6 +49,9 @@ vows.describe('escape').addBatch({
     'unescapes \'': function () {
       assert.strictEqual(unescapeXML('&apos;'), '\'')
     },
+    'leaves invalid entities alone': function () {
+      assert.strictEqual(unescapeXML('&foobar;'), '&foobar;')
+    },
```
Comment (sonnyp) [inline test/escape-test.js]: should we throw instead?

Comment (mogsie) [reply test/escape-test.js]: That was my first inclination, and my first implementation, but I didn't want to break backwards compatibility, so I decided to write a test that verified that the behaviour didn't change.  If anything, that might be better to put in a different PR.

Comment (sonnyp) [reply test/escape-test.js]: sounds sane :) good candidate for release 3.0

**Comment by sonnyp** [comment]
https://github.com/xmppjs/ltx/releases/tag/v2.8.0

**Pull request merged**